src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/ProjectKNuGetProject.cs

```diff
@@ -105,11 +89,25 @@ public ProjectKNuGetProject(INuGetPackageManager project, string projectName, st
                 throw new ArgumentException(ProjectManagement.Strings.PackageStreamShouldBeSeekable);
             }
 
+            // Uninstall the package if it is already installed. This should only happen when an
+            // update occurred from Install-Package PMC command, the Browse tab in the UI, or the
+            // Installed tab in the UI. An update from the Updates tab has an explicit Uninstall
+            // action before the install.
+            var installedPackages = await GetInstalledPackagesAsync(token);
+            var packageToReplace = installedPackages
+                .Where(pr => StringComparer.OrdinalIgnoreCase.Equals(pr.PackageIdentity.Id, packageIdentity.Id))
+                .FirstOrDefault();
+
+            if (packageToReplace != null)
+            {
+                await UninstallPackageAsync(packageToReplace.PackageIdentity, nuGetProjectContext, token);
```

Comment:
Correct. The `UninstallPackageAsync` is the exact same method that is invoked when during a batch update or an explicit uninstall.